### DISCUSSION TITLE
`straight-freeze-versions` produces massive changes in my version lockfile.  Now what?

### DISCUSSION THREAD

**Discussion by dabrahams** [raxod502/straight.el#1161] [Q&A]
I assume this means that the versions I have checked out on this machine apparently are way out of sync with the version lockfile.  I don't know how to resolve this situation.  I _think_ I'd prefer not to roll any versions in the lockfile back, and instead choose the later of the two options each place there's a change.  Any advice for me?  Is there a good way to avoid this situation in future?

**Comment by raxod502** [comment]
What do you mean by resolve the situation? Do you want to update your packages on this machine to match the lockfile, or do you want to update the lockfile to match what you have on this machine? If the former - you've used the wrong command, you want `M-x straight-thaw-versions`. If the latter - check the updated lockfile into version control and you're good to go.

If you just want to update every package to the latest available version, there is `M-x straight-pull-recipe-repositories`, `M-x straight-normalize-all`, `M-x straight-x-fetch-all`, `M-x straight-merge-all` for that, followed by writing a new lockfile. If you have some packages that you want to upgrade and others you want to downgrade... how did that happen?

**Comment by dabrahams** [comment]
> Do you want to update your packages on this machine to match the lockfile, or do you want to update the lockfile to match what you have on this machine? 

Well, I _was_ looking for guidance in this, but as I wrote above I guess the answer is "neither":

> I'd prefer not to roll any versions in the lockfile back, and instead choose the later of the two options each place there's a change

> If you have some packages that you want to upgrade and others you want to downgrade... how did that happen?

First of all, I have no idea how any of this happened.  My assumption was that if I have a lockfile, straight would ensure that the versions of packages I have are consistent with it.  Otherwise what good is the lockfile (and how appropriate is the name)?
 
I don't want to downgrade anything but it seemed safest to stick with versions I have used _somewhere_.

Updating everything to the latest version is an option.
